flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/ShardConsumer.java

```diff
@@ -96,6 +100,14 @@ protected ShardConsumer(KinesisDataFetcher<T> fetcherRef,
 							SequenceNumber lastSequenceNum,
 							KinesisProxyInterface kinesis) {
 		this.fetcherRef = checkNotNull(fetcherRef);
+		MetricGroup kinesisMetricGroup = fetcherRef.getRuntimeContext()
+			.getMetricGroup()
+			.addGroup("Kinesis");
+		kinesisMetricGroup
```

Comment:
use `addGroup("shard_id", subscribedShard.getShard().getShardId())` instead and register the metric on the returned group.